cpp/src/arrow/type.cc

```diff
@@ -79,6 +105,38 @@ std::shared_ptr<Field> Field::Copy() const {
   return ::arrow::field(name_, type_, nullable_, metadata_);
 }
 
+const std::string& Field::fingerprint() const {
+  if (!fingerprint_computed_) {
+    const auto& type_fingerprint = type_->fingerprint();
+    if (!type_fingerprint.empty()) {
+      std::stringstream ss;
```

Comment:
do stringstreams have some sort of small buffer allocated ahead of time (ala small string optimizations in standard strings?)

Reply:
I have no idea. I suspect it depends on the implementation.